lib/screens/translation_screen.dart

```diff
@@ -0,0 +1,360 @@
+        fontSize: 18.0,
+        color: Theme.of(context).primaryColorDark);
+
+    final controls = Container(
+        decoration: BoxDecoration(
+          color: Colors.white,
+          borderRadius: BorderRadius.circular(30.0),
+        ),
+        padding:
+            EdgeInsets.symmetric(horizontal: padding, vertical: padding / 2),
+        child:
+            Row(mainAxisAlignment: MainAxisAlignment.spaceBetween, children: [
+          Text(_sourceLanguage ?? "", style: textStyle),
+          _recorderWidget(),
+          Text(_targetLang ?? "", style: textStyle),
+        ]));
+    return Padding(padding: EdgeInsets.all(padding), child: controls);
+  }
+
+  Widget _timerWidget() {
+    return Text("$_recorderTimer", style: TextStyle(fontSize: 30.0));
+  }
+
+  Widget _buildSentencePageWidget(data) {
+    final itemList = PageView.builder(
+      controller: _pageController,
+      itemCount: data.length,
+      itemBuilder: (context, index) {
+        Sentence item = data[index];
+        return _buildSentenceCard(item);
+      },
+      onPageChanged: (index) {
+        setState(() {
+          _currentSentence = data[index];
+          _recorderTimer = "";
+          _progression = index + 1;
+        });
+      },
+    );
+
+    return Container(child: itemList, padding: EdgeInsets.all(16.0));
+  }
+
+  Widget _buildSentenceCard(Sentence phrase) {
+    return Card(
+        elevation: 2.0,
+        child: Container(
+            height: MediaQuery.of(context).size.height / 2,
+            width: MediaQuery.of(context).size.width,
+            padding: EdgeInsets.all(padding),
+            child: Center(
+                child: Text(
+              phrase.text,
+              style: TextStyle(
+                  color: Theme.of(context).accentColor,
+                  letterSpacing: 0.75,
+                  fontSize: 16.0,
+                  fontWeight: FontWeight.bold),
+            ))));
+  }
+
+  Widget _loadSentencesWidget() {
+    return FutureBuilder(
+        future: _fetchAndCacheData(),
+        builder: (context, snapshot) {
+          if (!snapshot.hasError && snapshot.hasData) {
+            _sentences = snapshot.data;
+            return _buildSentencePageWidget(_sentences);
+          } 
+          else {
+            return Center(child: CircularProgressIndicator());
+          }
+        });
+  }
+
+  Widget _progressionBar() {
+    int size = (_sentences != null) ? _sentences.length : 1;
+    return LinearProgressIndicator(value: _progression / size);
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _recorderTimer = "";
+    _isRecording = false;
+    _flutterSound = new FlutterSound();
+    _pageController = PageController();
+
+    // Create App Recording Folder
+    new Directory(recordStorage).create();
+
+    _getUserPreferences().then((prefs) {
+      setState(() {
+        setState(() {
+          this._author = prefs[0];
+          this._sourceLanguage = prefs[1];
```

Comment:
Sometimes to access preferences we use magic numbers, other times magic strings, I think we should stick with one and it should be strings. As you already know, I am not a fan of everything magic thus a better solution might be to have an enum for it as well.